File: solutions/javascript/resistor-color-duo/1/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const decodedValue = (colors) => {

  return parseInt(colors.slice(0, 2).reduce((acc, color, id)=>{
      return acc+= COLORS.indexOf(color);
  }, ''))
};

const COLORS=[
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
]

File: solutions/go/resistor-color-duo/1/resistor_color_duo.go
package resistorcolorduo

import (
	"fmt"
    "strconv"
)

func IndexOf(arr []string, searchedValue string) int{
    for index, val:= range arr{
        if searchedValue==val{
            return index
        }
    }
    return -1
}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
    colorsValues := []string{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}
    res := ""
    for index, color := range colors{
        res+=fmt.Sprintf("%d", IndexOf(colorsValues, color))
        if index==1{
            colorValue, err := strconv.Atoi(res)
            if err != nil {
                // ... handle error
                panic(err)
            }
            return int(colorValue)
        }
    }
	return 0
}
